### PR TITLE
nmd, nmt: init at 0.5.0

### PR DESCRIPTION
**Pull request by rycee**
## Description of changes

Adds nmt and nmd as packages. See https://github.com/nix-community/home-manager/issues/4879 for motivation.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ni/nix-lib-nmt/package.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  };
+
+  installPhase = ''
+    mkdir -pv "$out"
+    cp -rv * "$out"
+  '';
+
+  meta = {
+    homepage = "https://git.sr.ht/~rycee/nmt";
+    description =
```
Comment (kirillrdy) [inline pkgs/by-name/ni/nix-lib-nmt/package.nix:28]: nit pick: given there is currently not agreed formatting, any reason for splitting lines ?

File: pkgs/by-name/ni/nix-lib-nmt/package.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      "A basic test framework for projects using the Nixpkgs module system";
```
Comment (kirillrdy) [inline pkgs/by-name/ni/nix-lib-nmt/package.nix:29]:
```suggestion
      "A basic test framework for projects using the nixpkgs module system";
```

File: pkgs/by-name/ni/nix-lib-nmt/package.nix (hunk 1 above)
Comment (rycee) [reply pkgs/by-name/ni/nix-lib-nmt/package.nix]: I just ran nixfmt on the file and that's what it suggested. Joined the lines now.

File: pkgs/by-name/ni/nix-lib-nmt/package.nix (hunk 2 above)
Comment (rycee) [reply pkgs/by-name/ni/nix-lib-nmt/package.nix]: Nixpkgs is a proper noun and seems to be referred to with a capital N in many places. See, e.g., the repo README.md file.

**Comment by rycee** [comment]
Thanks for having a look @kirillrdy 🙂

File: pkgs/by-name/ni/nix-lib-nmt/package.nix (hunk 1 above)
Comment (kirillrdy) [reply pkgs/by-name/ni/nix-lib-nmt/package.nix]: fair enough, i use `nixpkgs-fmt` I personally don't mind, just haven't seen line split before

File: pkgs/by-name/ni/nix-lib-nmt/package.nix (hunk 2 above)
Comment (kirillrdy) [reply pkgs/by-name/ni/nix-lib-nmt/package.nix]: ok thats a fair call, even though I would expected `nixpkgs`, since we type is so damn frequently

**Review by kirillrdy — APPROVED** [review]
Result of `nixpkgs-review pr 280467` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>nix-lib-nmd</li>
    <li>nix-lib-nmt</li>
  </ul>
</details>

**Comment by kirillrdy** [comment]
@rycee feel free to ping me for reviews

**Pull request merged**